cloudinit/config/cloud-init-schema.json

```diff
@@ -783,6 +783,97 @@
         }
       }
     },
+    "cc_mcollective": {
+      "type": "object",
+      "properties": {
+        "mcollective": {
+          "type": "object",
+          "properties": {
+            "conf": {
+              "type": "object",
+              "properties": {
+                "public-cert": {
+                  "type": "string",
+                  "description": "Optional value of server public certificate which will be written to ``/etc/mcollective/ssl/server-public.pem``"
+                },
+                "private-cert": {
+                  "type": "string",
+                  "description": "Optional value of server private certificate which will be written to ``/etc/mcollective/ssl/server-private.pem``"
+                }
+              },
+              "patternProperties": {
```

Comment:
Do `properties` and `patternProperties` co-exist at the same level peacefully? Can we add a test where we set public-cert or private-cert to a boolean or integer and verify it fails to show that the `patternProperties` definition isn't replacing the `properties` definition?

Reply:
Yes they do. I had wondered the same things, which is why we [already have a test for explicit property types like public-cert in the unittests of this PR](https://github.com/canonical/cloud-init/pull/1358/files#diff-226b092ec1d5779ac38368bed2be8bb6a67701d48155e01604039169005906c2R176-R178)